src/en/juju-upgrade.md

```diff
@@ -0,0 +1,191 @@
+Title: Upgrading Juju software
+
+
+# Upgrading Juju software
+
+At the highest level, a Juju topology can be divided into two main parts:
+
+- Juju client
```

Comment:
This stuff will be meshed/linked with an upcoming architectural overview (see issue #662) but, yes, at this time I will add such a note.